### PR TITLE
Appying PEP 621: storing metadata in `pyproject.toml`

### PR DESCRIPTION
**Pull request by zliang-akamai**
## Description
The setuptools support is still in beta phase. Hope we can adopt this by the end of this year.

## Testing
- You can do `make create-version` to verify the version file is created correctly.
- After creating the version,`make dev-install` to verify all dependencies and dev dependencies are installed.
- Write any code to verify the SDK is working.

For example:
```python
from linode_api4 import LinodeClient, Instance

client = LinodeClient("MY_PERSONAL_ACCESS_TOKEN")

new_linode, root_pass = client.linode.instance_create(
    ltype="g6-nanode-1",
    region="us-mia",
    image="linode/ubuntu22.04",
    label="my-ubuntu-linode"
)

new_linode.delete()
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,70 @@
...
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Libraries",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
```
Comment (lgarber-akamai) [inline pyproject.toml:31]: Would it make sense to also add Python 3.12 here?

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,70 @@
 [build-system]
-requires = ["setuptools"]
+requires = ["setuptools", "wheel"]
 build-backend = "setuptools.build_meta"
 
+
+[project]
+name = "linode_api4"
+authors = [{ name = "Linode", email = "developers@linode.com" }]
+description = "The official python SDK for Linode API v4"
```
Comment (lgarber-akamai) [inline pyproject.toml:9]: Nitpick from the original metadata but we might as well fix it in this PR 👍 

```suggestion
description = "The official Python SDK for Linode API v4"
```

File: Makefile [hunk 3]
```diff
@@ -12,60 +16,63 @@ ifdef TEST_MODEL
...
-@PHONEY: clean
+.PHONY: clean
 clean:
 	mkdir -p dist
 	rm -r dist
 	rm -f baked_version
 
-@PHONEY: build
+.PHONY: build
 build: clean
```
Comment (lgarber-akamai) [inline Makefile:26]: Would it make sense for the `build` target to depend on the `create-version` target?

File: Makefile [hunk 4]
```diff
@@ -12,60 +16,63 @@ ifdef TEST_MODEL
...
+.PHONY: dev-install
+dev-install: clean
+	$(PYTHON) -m pip install -e ".[dev]"
 
-@PHONEY: requirements
-requirements:
-	$(PYTHON) -m pip install -r requirements.txt -r requirements-dev.txt
+.PHONY: install
+install: clean
+	$(PYTHON) -m pip install .
```
Comment (lgarber-akamai) [inline Makefile:43]: Same feedback as the build target

File: pyproject.toml (hunk 1 above)
Comment (zliang-akamai) [reply pyproject.toml:31]: Just added, thanks!

**Review by ykim-1 — APPROVED** [review]
Looks good! Verified the install steps locally

**Pull request merged**